### ISSUE TITLE
Mis-specification of `portalocker` dependency

### ISSUE THREAD

**Issue by dhirschfeld** [conda-forge/msal_extensions-feedstock#11]
The `portalocker` dep doesn't reflect the upstream version constraints:
https://github.com/conda-forge/msal_extensions-feedstock/blob/fccb9f9ed5c0ca268ad0667a819a71326dc41017/recipe/meta.yaml#L24
* https://github.com/AzureAD/microsoft-authentication-extensions-for-python/blob/dev/setup.py#L25-L26

This causes `pip check` to fail on all CF repos which depend on this repo as they install the latest [`2.3.0`](https://github.com/WoLpH/portalocker/releases/tag/v2.3.0) version of `portalocker`

e.g. 
```
import: 'azure.identity'
import: 'azure.identity._credentials'
import: 'azure.identity'
import: 'azure.identity._credentials'
+ pip check
msal-extensions 0.3.0 has requirement portalocker~=1.0; platform_system != "Windows", but you have portalocker 2.3.0.
Tests failed for azure-identity-1.6.1-pyhd8ed1ab_0.tar.bz2 - moving package to /home/conda/feedstock_root/build_artifacts/broken
```

**Comment by dhirschfeld** [comment]
***xref:***
* https://github.com/conda-forge/azure-identity-feedstock/pull/6#issuecomment-902377198
* https://github.com/conda-forge/adlfs-feedstock/pull/47#issuecomment-900883604

**Comment by maresb** [comment]
I just [ran into](https://dev.azure.com/conda-forge/feedstock-builds/_build/results?buildId=413319&view=logs&j=171a126d-c574-5c8c-1269-ff3b989e923d&t=1183ba29-a0b5-5324-8463-2a49ace9e213&l=2024)
```
msal-extensions 0.3.0 has requirement portalocker~=1.6; platform_system == "Windows", but you have portalocker 1.4.0.
```
because the build 0 was being installed instead of build 1. I think we need to [make a repodata patch](https://github.com/conda-forge/conda-forge-repodata-patches-feedstock/tree/master/recipe) to fix build 0.

**Comment by h-vetinari** [comment]
This is still current - now running into this in https://github.com/conda-forge/torchdata-feedstock/pull/35 with the same message:
```
+ pip check
msal-extensions 0.3.0 has requirement portalocker~=1.0; platform_system != "Windows", but you have portalocker 3.0.0.
```

**Comment by h-vetinari** [comment]
This was fixed in https://github.com/AzureAD/microsoft-authentication-extensions-for-python/pull/97, so the question arises why none of the version updates since then were merged (e.g. #28)

**Comment by dhirschfeld** [comment]
Probably just fell off my radar - I've been accumulating a large backlog of things I need to attend to 😬 

Anyway, merged that PR so 1.2.0 is now available.

**Comment by h-vetinari** [comment]
This should be fixed as of #28, which adds a `pip check` - thanks!

**Comment by dhirschfeld** [comment]
I'll go ahead and close this. It can always be reopened if there are still issues...
